lib/site_prism/dsl.rb

```diff
@@ -228,27 +222,31 @@ def create_error_method(name)
       end
 
       def deduce_iframe_scope_find_args(args)
+        warn_on_invalid_selector_input(args)
         case args[0]
-        when Integer
-          [args[0]]
-        when String
-          [:css, args[0]]
-        else
-          args
+        when Integer then [args[0]]
+        when String  then [:css, args[0]]
+        else args
         end
       end
 
       def deduce_iframe_element_find_args(args)
+        warn_on_invalid_selector_input(args)
         case args[0]
-        when Integer
-          "iframe:nth-of-type(#{args[0] + 1})"
-        when String
-          [:css, args[0]]
-        else
-          args
+        when Integer then "iframe:nth-of-type(#{args[0] + 1})"
+        when String  then [:css, args[0]]
+        else args
         end
       end
 
+      def warn_on_invalid_selector_input(args)
+        return unless !args[0].is_a?(String) || !args[0].start_with?('/', './')
```

Comment:
It might be re-written like this
`return if args[0].is_a?(String) && args[0].start_with?('/', './')`
Can `args[0]` be `nil` by mistake? We should provide a meaningful message than

And it will warn if `args[0]` is not a string, which is not expected behavior. 

The condition should be something like
`return if !args[0].is_a?(String) || args[0].start_with?('/', './')`

Reply:
If it's `nil` it gets caught in the `build` phase and will create error methods for everything and crazy lazily. So we're fine with that 2nd condition (I've fixed up the first bit as per DeMorgans Law)